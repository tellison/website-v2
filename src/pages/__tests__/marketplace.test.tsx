import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe';
import Marketplace, { Head } from '../marketplace';

describe('Slack page', () => {
  it('renders correctly', () => {
    const { container } = render(<Marketplace />);
    // eslint-disable-next-line
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('head renders correctly', () => {
    const { container } = render(<Head />);
    // eslint-disable-next-line
    const title = container.querySelector('title');
    expect(title?.textContent).toEqual('Marketplace | Adoptium');
  });

  it('has no accessibility violations', async () => {
    const { container } = render(<Marketplace />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
