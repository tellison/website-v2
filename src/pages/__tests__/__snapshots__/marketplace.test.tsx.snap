// Vitest Snapshot v1

exports[`Slack page > renders correctly 1`] = `
<main>
  <section
    class="py-5 text-center container"
  >
    <div
      class="row py-lg-5"
    >
      <div
        class="col-lg-8 col-md-8 mx-auto"
      >
        <h1
          class="fw-light"
        >
          AdoptiumÂ® Marketplace
        </h1>
        <p
          class="pt-3"
        >
          Text
        </p>
        <div
          class="row align-items-center pt-3"
        >
          <div
            class="col-6 col-md-4"
          >
            <img
              alt="AQAvit logo"
              class="img-fluid"
              src="/images/aqavit-light.png"
              width="150"
            />
          </div>
          <div
            class="col-12 col-sm-6 col-md-8 "
          >
            <p
              class="text-start"
            >
              Text
            </p>
          </div>
        </div>
      </div>
    </div>
    <ul
      class="vendor-list pt-5"
    >
      <li
        class="vendor-li"
      >
        <input
          checked=""
          id="vendorAdoptium"
          type="checkbox"
        />
        <label
          class="vendor-label"
          for="vendorAdoptium"
          title="Adoptium"
        >
          <img
            alt="Adoptium icon"
            src="/images/vendors/vendor-adoptium.png"
          />
        </label>
      </li>
      <li
        class="vendor-li"
      >
        <input
          checked=""
          id="vendorRed Hat"
          type="checkbox"
        />
        <label
          class="vendor-label"
          for="vendorRed Hat"
          title="Red Hat"
        >
          <img
            alt="Red Hat icon"
            src="/images/vendors/vendor-redhat.png"
          />
        </label>
      </li>
      <li
        class="vendor-li"
      >
        <input
          checked=""
          id="vendorMicrosoft"
          type="checkbox"
        />
        <label
          class="vendor-label"
          for="vendorMicrosoft"
          title="Microsoft"
        >
          <img
            alt="Microsoft icon"
            src="/images/vendors/vendor-microsoft.png"
          />
        </label>
      </li>
      <li
        class="vendor-li"
      >
        <input
          checked=""
          id="vendorAzul"
          type="checkbox"
        />
        <label
          class="vendor-label"
          for="vendorAzul"
          title="Azul"
        >
          <img
            alt="Azul icon"
            src="/images/vendors/vendor-azul.png"
            style="padding: 0.7em;"
          />
        </label>
      </li>
      <li
        class="vendor-li"
      >
        <input
          checked=""
          id="vendorIBM"
          type="checkbox"
        />
        <label
          class="vendor-label"
          for="vendorIBM"
          title="IBM"
        >
          <img
            alt="IBM icon"
            src="/images/vendors/vendor-ibm.png"
          />
        </label>
      </li>
      <li
        class="vendor-li"
      >
        <input
          checked=""
          id="vendorHuawei"
          type="checkbox"
        />
        <label
          class="vendor-label"
          for="vendorHuawei"
          title="Huawei"
        >
          <img
            alt="Huawei icon"
            src="/images/vendors/vendor-huawei.png"
          />
        </label>
      </li>
      <li
        class="vendor-li"
      >
        <input
          checked=""
          id="vendorAlibaba"
          type="checkbox"
        />
        <label
          class="vendor-label"
          for="vendorAlibaba"
          title="Alibaba"
        >
          <img
            alt="Alibaba icon"
            src="/images/vendors/vendor-alibaba.png"
          />
        </label>
      </li>
    </ul>
    <div
      class="input-group mb-5 row g-2"
    >
      <div
        class="input-group-prepend flex-colunm col-12 col-md-3"
      >
        <label
          class="px-2 fw-bold"
          for="os"
        >
          Text
        </label>
        <select
          aria-label="OS Filter"
          class="form-select form-select-sm"
          data-testid="os-filter"
          id="os-filter"
        >
          <option
            value="any"
          >
            Any
          </option>
          <option
            value="linux"
          >
            Linux
          </option>
          <option
            value="alpine-linux"
          >
            Alpine Linux
          </option>
          <option
            value="windows"
          >
            Windows
          </option>
          <option
            value="mac"
          >
            macOS
          </option>
          <option
            value="aix"
          >
            AIX
          </option>
          <option
            value="solaris"
          >
            Solaris
          </option>
        </select>
      </div>
      <div
        class="input-group-prepend flex-colunm col-12 col-md-3"
      >
        <label
          class="px-2 fw-bold"
          for="arch"
        >
          Text
        </label>
        <select
          aria-label="Architecture Filter"
          class="form-select form-select-sm"
          data-testid="arch-filter"
          id="arch-filter"
        >
          <option
            value="any"
          >
            Any
          </option>
          <option
            value="x64"
          >
            x64
          </option>
          <option
            value="x86"
          >
            x86
          </option>
          <option
            value="aarch64"
          >
            aarch64
          </option>
          <option
            value="s390x"
          >
            s390x
          </option>
          <option
            value="ppc64le"
          >
            ppc64le
          </option>
          <option
            value="ppc64"
          >
            ppc64
          </option>
          <option
            value="arm"
          >
            arm
          </option>
          <option
            value="sparcv9"
          >
            sparcv9
          </option>
        </select>
      </div>
      <div
        class="input-group-prepend flex-colunm col-12 col-md-3"
      >
        <label
          class="px-2 fw-bold"
          for="package-type"
        >
          Text
        </label>
        <select
          aria-label="Package Type Filter"
          class="form-select form-select-sm"
          data-testid="package-type-filter"
          id="package-type-filter"
        >
          <option
            value="any"
          >
            Any
          </option>
          <option
            value="jdk"
          >
            JDK
          </option>
          <option
            value="jre"
          >
            JRE
          </option>
        </select>
      </div>
      <div
        class="input-group-prepend flex-colunm col-12 col-md-3"
      >
        <label
          class="px-2 fw-bold"
          for="version"
        >
          Text
        </label>
        <select
          aria-label="Version Filter"
          class="form-select form-select-sm"
          data-testid="version-filter"
          id="version-filter"
        >
          <option
            value="17"
          >
            17
          </option>
          <option
            value="11"
          >
            11
          </option>
          <option
            value="8"
          >
            8
          </option>
        </select>
      </div>
    </div>
    <table
      class="table table-bordered releases-table"
      id="download-table"
      style="border-spacing: 0 10px; border-collapse: separate;"
    >
      <thead
        class="table-dark"
      >
        <tr
          class="table-head"
        >
          <td
            class="table-head"
          >
            Text
          </td>
          <td
            class="table-head"
          >
            Text
          </td>
          <td
            class="table-head"
          >
            Text
          </td>
          <td
            class="table-head"
          >
            Text
          </td>
          <td
            class="table-head"
          >
            Text
          </td>
          <td
            class="table-head"
          >
            Text
          </td>
        </tr>
      </thead>
      <tbody
        class="table-light"
      >
        <tr />
      </tbody>
    </table>
    <div
      aria-hidden="true"
      aria-labelledby="checksumModalLabel"
      class="modal fade"
      id="checksumModal"
      style="z-index: 10000;"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-lg"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h5
              class="modal-title"
              id="checksumModalLabel"
            >
              Checksum (SHA256)
            </h5>
            <button
              aria-label="Close"
              class="btn-close"
              data-bs-dismiss="modal"
              type="button"
            />
          </div>
          <div
            class="modal-body text-start"
          >
            <p>
              The checksum is used to ensure the file has not been corrupted during download. You should compare the checksum of the file you received with the value below to ensure the file you received is complete and unmodified.
            </p>
            <p>
              On Windows use 
              <code>
                certUtil -hashfile 
                <var>
                  file
                </var>
                 SHA256
              </code>
              , on Linux use 
              <code>
                sha256sum 
                <var>
                  file
                </var>
              </code>
              , and on macOS use 
              <code>
                shasum -a 256 
                <var>
                  file
                </var>
              </code>
              .
            </p>
            <input
              readonly=""
              style="width: 100%;"
              value=""
            />
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Copy
            </button>
            <button
              class="btn btn-secondary"
              data-bs-dismiss="modal"
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</main>
`;
